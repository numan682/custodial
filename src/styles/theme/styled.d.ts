import 'styled-components'
import type { ThemeType } from 'grommet/es6/themes'

/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends ThemeType {}
}
