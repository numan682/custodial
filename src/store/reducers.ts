/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { AnyAction, combineReducers } from '@reduxjs/toolkit'
import createWalletReducer from 'app/pages/CreateWalletPage/slice'
import fiatOnrampReducer from 'app/pages/FiatOnrampPage/slice'
import accountReducer from 'app/state/account'
import fatalErrorReducer from 'app/state/fatalerror'
import importAccountsReducer from 'app/state/importaccounts'
import networkReducer from 'app/state/network'
import paraTimesReducer from 'app/state/paratimes'
import stakingReducer from 'app/state/staking'
import transactionReducer from 'app/state/transaction'
import walletReducer from 'app/state/wallet'
import themeReducer from 'styles/theme/slice'
import persistReducer, { persistActions, receivePersistedRootState } from 'app/state/persist'
import { RootState } from 'types'

function createRootReducer() {
  const rootReducer = combineReducers({
    account: accountReducer,
    createWallet: createWalletReducer,
    fiatOnramp: fiatOnrampReducer,
    fatalError: fatalErrorReducer,
    importAccounts: importAccountsReducer,
    network: networkReducer,
    paraTimes: paraTimesReducer,
    staking: stakingReducer,
    theme: themeReducer,
    transaction: transactionReducer,
    wallet: walletReducer,
    persist: persistReducer,
  })

  return rootReducer
}

export function createPersistedRootReducer() {
  const originalRootReducer = createRootReducer()
  return (state: RootState | undefined, action: AnyAction): RootState => {
    const newState = originalRootReducer(state, action)
    if (persistActions.setUnlockedRootState.match(action)) {
      return receivePersistedRootState(
        newState,
        action.payload.persistedRootState,
        action.payload.stringifiedEncryptionKey,
      )
    }
    if (persistActions.resetRootState.match(action)) {
      return originalRootReducer(undefined, action)
    }
    return newState
  }
}
