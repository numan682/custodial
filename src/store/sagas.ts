import { all } from 'typed-redux-saga'

import { accountSaga } from 'app/state/account/saga'
import { networkSaga } from 'app/state/network/saga'
import { importAccountsSaga } from 'app/state/importaccounts/saga'
import { paraTimesSaga } from 'app/state/paratimes/saga'
import { stakingSaga } from 'app/state/staking/saga'
import { transactionSaga } from 'app/state/transaction/saga'
import { rootWalletSaga } from 'app/state/wallet/saga'
import { persistSaga } from 'app/state/persist/saga'

export default function* rootSagas() {
  yield* all([
    accountSaga(),
    networkSaga(),
    importAccountsSaga(),
    paraTimesSaga(),
    stakingSaga(),
    transactionSaga(),
    rootWalletSaga(),
    persistSaga(),
  ])
}
