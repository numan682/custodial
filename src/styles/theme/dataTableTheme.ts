import { IDataTableStyles, ITheme } from 'react-data-table-component'

export const dataTableTheme = {
  background: { default: 'unset' },
  highlightOnHover: {
    default: '#88888833',
    text: 'unset',
  },
  text: {
    primary: 'unset',
    secondary: 'unset',
    disabled: 'unset',
  },
  sortFocus: {
    default: 'unset',
  },
  divider: { default: '#AAAAAAaa' },
} as Partial<ITheme>

export const dataTableCustomStyles: IDataTableStyles = {
  headCells: {
    style: {
      fontSize: '18px',
    },
  },
  rows: {
    style: {
      fontSize: '16x',
    },
  },
}
