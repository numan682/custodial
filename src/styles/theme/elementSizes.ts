export const sidebarSmallSizeLogo = 30
export const sidebarMediumSizeLogo = 40
export const mobileToolbarZIndex = 3
export const mobileToolbarButtonsZIndex = mobileToolbarZIndex + 1
export const buildBannerZIndex = mobileToolbarZIndex - 1

export const mobileFooterNavigationHeight = '4rem'
