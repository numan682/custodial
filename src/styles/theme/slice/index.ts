import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'

import { getThemeFromStorage, saveTheme } from '../utils'
import { ThemeState } from './types'

export const getInitialState = (): ThemeState => ({
  selected: getThemeFromStorage() || 'system',
})

const slice = createSlice({
  name: 'theme',
  initialState: () => getInitialState(),
  reducers: {
    changeTheme(state, action: PayloadAction<'dark' | 'light'>) {
      saveTheme(action.payload)
      state.selected = action.payload
    },
  },
})

export const { actions: themeActions } = slice

export default slice.reducer
