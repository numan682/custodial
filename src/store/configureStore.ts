import createSagaMiddleware from 'redux-saga'

import { createPersistedRootReducer } from './reducers'
import rootSagas from './sagas'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'
import { configureStoreWithSyncTabs } from 'app/state/persist/syncTabs'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.toString(),
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware]

  const store = configureStoreWithSyncTabs({
    reducer: createPersistedRootReducer(),
    middleware: getDefaultMiddleware => getDefaultMiddleware().concat(middlewares),
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production',
    preloadedState: state,
  })

  sagaMiddleware.run(rootSagas)

  return store
}
