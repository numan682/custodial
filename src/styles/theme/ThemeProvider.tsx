import { Grommet } from 'grommet/es6/components/Grommet'
import { grommet, ThemeType } from 'grommet/es6/themes'
import { deepMerge } from 'grommet/es6/utils'
import * as React from 'react'
import { createTheme as dataTableCreateTheme } from 'react-data-table-component'
import { useSelector } from 'react-redux'

import { selectTheme } from './slice/selectors'
import { dataTableTheme } from './dataTableTheme'

/**
 * React-data-table by default sets its own background and text colors
 * we make sure that they do not override grommet's
 */
dataTableCreateTheme('blank', dataTableTheme)

const grommetCustomTheme: ThemeType = {
  button: {
    primary: {
      background: {
        dark: 'red',
        light: 'green',
      },
    },
    border: {
      radius: '4px',
    },
    badge: {
      size: { medium: '20px' },
      container: {
        extend: `
          margin-top: -10px;
          margin-right: -10px;
        `,
      },
    },
  },
  tip: {
    content: {
      // Default background is background-contrast, but we made that one transparent
      background: 'background-front',
      border: 'all',
      elevation: 'none',
    },
  },
  select: {
    options: {
      container: {
        // Remove padding to match ParaTimeOption when displayed as value and as option
        pad: 'none',
      },
    },
  },
  global: {
    colors: {
      oasisIndigo: '#310081',
      oasisMinty: '#4CD4A9',
      oasisLightGray: '#ececec',
      brand: {
        dark: '#0092f6bb',
        light: '#0092f6',
      },
      oasisBlue2: '#4db3f9',
      oasisBlue3: '#26a2f8',
      'status-ok': '#2ad5ab',
      'status-warning': {
        dark: '#f3d45e',
        light: '#e1a809',
      },
      'status-error': '#d24c00',

      'alert-box-info': {
        dark: '#0092f6',
        light: '#d4ebff',
      },
      'alert-box-info-background': {
        dark: '#d4ebff',
        light: '#0092f6',
      },
      'alert-box-ok-weak': {
        dark: '#2ad5ab',
        light: '#2ad5ab',
      },
      'alert-box-ok-weak-background': {
        dark: '#17464b',
        light: '#c2f5ea',
      },
      'alert-box-ok': {
        dark: '#2ad5ab',
        light: '#c2f5ea',
      },
      'alert-box-ok-background': {
        dark: '#c2f5ea',
        light: '#2ad5ab',
      },
      'alert-box-warning': {
        dark: '#f3d45e',
        light: '#f0e8cb',
      },
      'alert-box-warning-background': {
        dark: '#f0e8cb',
        light: '#f3d45e',
      },
      'alert-box-error': {
        dark: '#f26111',
        light: '#ffe7d9',
      },
      'alert-box-error-background': {
        dark: '#ffe7d9',
        light: '#f26111',
      },

      'background-oasis-blue': {
        dark: '#0f477b',
        light: '#0092f6',
      },
      lightText: '#a3a3a3',
      neutral: {
        dark: '#310081FF',
        light: '#310081FF',
      },
      'neutral-2': {
        dark: '#0092f6bb',
        light: '#0092f6bb',
      },
      'background-back': {
        dark: '#1A1A2e',
        light: '#EFEFEF',
      },
      'background-front': {
        dark: '#16213e',
        light: '#FFFFFF',
      },
      'background-front-lighter': {
        dark: '#222938',
        light: '#FFFFFF',
      },
      'background-front-border': {
        dark: '#111111',
        light: '#EDEDED',
      },
      'background-contrast': {
        dark: '#FFFFFF08',
        light: '#11111108',
      },
      'background-contrast-2': {
        dark: '#FFFFFF33',
        light: '#11111133',
      },
      'background-custom': {
        dark: '#0E5265',
        light: '#00C8FF',
      },
      'component-toolbar': {
        dark: '#0f346088',
        light: '#26a2f844',
      },
      'component-sidebar': {
        dark: '#16213e',
        light: '#fafafa',
      },
      'successful-label': {
        dark: '#6FFFB0',
        light: '#3fa900',
      },

      text: {
        dark: '#F8F8F8',
        light: '#16213E',
      },
    },
    font: {
      family: 'Rubik',
    },
  },
  notification: {
    toast: {
      time: 2000,
      container: {
        border: 'all',
        elevation: 'none',
      },
    },
  },
}
export const ThemeProvider = (props: { children: React.ReactChild }) => {
  const theme = deepMerge(grommet, grommetCustomTheme)
  const mode = useSelector(selectTheme)

  return (
    <Grommet theme={theme} themeMode={mode} style={{ minHeight: '100vh' }}>
      {React.Children.only(props.children)}
    </Grommet>
  )
}
