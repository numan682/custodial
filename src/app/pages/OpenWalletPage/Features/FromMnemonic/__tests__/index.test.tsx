import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { configureAppStore } from 'store/configureStore'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { FromMnemonic } from '..'

const renderPage = (store: any) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <FromMnemonic />
      </ThemeProvider>
    </Provider>,
  )

describe('<FromMnemonic/>', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  // @todo this has a random mnemonic...
  it('should match snapshot', () => {
    const page = renderPage(store)
    expect(page.container.firstChild).toMatchSnapshot()
  })

  it('should display an error on invalid mnemonic', async () => {
    renderPage(store)
    const textbox = screen.getByRole('textbox') as HTMLInputElement
    const button = screen.getByRole('button')
    await userEvent.type(textbox, 'hello')
    await userEvent.click(button)
    const errorElem = screen.getByText('openWallet.mnemonic.error')
    expect(errorElem).toBeInTheDocument()

    // A valid phrase should remove the error
    await userEvent.clear(textbox)
    await userEvent.type(textbox, 'echo toward hold roast rather reduce cute civil equal whale wait conduct')
    await userEvent.click(button)
    expect(errorElem).not.toBeInTheDocument()
  })

  it('newlines in mnemonic should be valid', async () => {
    renderPage(store)
    const textbox = screen.getByRole('textbox')
    const button = screen.getByRole('button', { name: 'openWallet.mnemonic.import' })
    await userEvent.type(
      textbox,
      'echo\ntoward hold   roast\n rather reduce cute civil equal whale wait conduct',
    )
    await userEvent.click(button)
    const errorElem = screen.queryByText('openWallet.mnemonic.error')
    expect(errorElem).toBeNull()
  })

  it('should display account selection modal window', async () => {
    renderPage(store)
    const textbox = screen.getByRole('textbox') as HTMLInputElement
    const button = screen.getByRole('button', { name: 'openWallet.mnemonic.import' })

    await userEvent.type(textbox, 'echo toward hold roast rather reduce cute civil equal whale wait conduct')
    await userEvent.click(button)

    expect(await screen.findByText('openWallet.importAccounts.selectWallets')).toBeInTheDocument()
  })
})
