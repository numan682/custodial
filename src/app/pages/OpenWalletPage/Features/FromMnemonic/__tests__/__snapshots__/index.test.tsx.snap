// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FromMnemonic/> should match snapshot 1`] = `
.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 12px;
  background-color: #FFFFFF;
  color: #16213E;
  border: solid 1px #EDEDED;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 5px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #11111108;
  color: #16213E;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 6px;
  background-color: #11111108;
  color: #16213E;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 6px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-right: 12px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(14ch,1fr) );
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(100%,1fr) );
  grid-gap: 12px 12px;
  padding: 0px;
}

.c14 {
  font-size: 18px;
  line-height: 24px;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #0092f6;
  color: #F8F8F8;
  border-radius: 4px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c9:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c9:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible) > circle,.c9:focus:not(:focus-visible) > ellipse,
.c9:focus:not(:focus-visible) > line,.c9:focus:not(:focus-visible) > path,
.c9:focus:not(:focus-visible) > polygon,.c9:focus:not(:focus-visible) > polyline,
.c9:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
}

.c3 {
  margin-top: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  font-size: 18px;
  line-height: 24px;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  height: 100%;
  max-width: 100%;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus > circle,
.c7:focus > ellipse,
.c7:focus > line,
.c7:focus > path,
.c7:focus > polygon,
.c7:focus > polyline,
.c7:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus::-moz-focus-inner {
  border: 0;
}

.c7::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c7::-moz-placeholder {
  color: #AAAAAA;
}

.c7:-ms-input-placeholder {
  color: #AAAAAA;
}

.c7::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c7::-moz-focus-inner {
  border: none;
  outline: none;
}

.c7:-moz-placeholder,
.c7::-moz-placeholder {
  opacity: 1;
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    border: solid 1px #EDEDED;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    margin-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c12 {
    margin: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c12 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c12 {
    padding: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

<div
  class="c0"
  style="min-height: 100vh;"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <form
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        spellcheck="false"
      >
        <h1
          class="c3"
        >
          openWallet.mnemonic.header
        </h1>
        <p
          class="c4"
        >
          openWallet.mnemonic.instruction
        </p>
        <div
          class="c5 "
        >
          <div
            class="c6 "
          >
            <textarea
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="c7"
              id="mnemonic"
              placeholder="openWallet.mnemonic.enterPhraseHere"
              rows="5"
              spellcheck="false"
            />
          </div>
        </div>
        <div
          class="c8"
        >
          <button
            class="c9"
            type="submit"
          >
            openWallet.mnemonic.import
          </button>
        </div>
      </form>
      <div
        class="c10"
      >
        <span
          class="notranslate"
          translate="no"
        >
          <div
            class="c11"
            data-testid="mnemonic-grid"
          >
            <div
              class="c12"
            >
              <div
                class="c13"
              >
                <span
                  class="c14"
                  style="user-select: none;"
                >
                  1
                  .
                </span>
              </div>
              <div
                class="c15"
              >
                <span
                  class="notranslate"
                  translate="no"
                >
                  <strong
                    style="white-space: pre; text-decoration: red wavy underline;"
                  />
                </span>
              </div>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;
